pprof/pprof.go

```diff
@@ -0,0 +1,51 @@
+package pprof
+
+import (
+	"context"
+	"github.com/NethermindEth/juno/utils"
+	"net/http"
+	_ "net/http/pprof"
+)
+
+type Pprof struct {
+	enabled bool
+	port    string
+	log     utils.SimpleLogger
+}
+
+func New(enabled bool, port string, log utils.SimpleLogger) *Pprof {
+	return &Pprof{
+		enabled: enabled,
+		port:    port,
+		log:     log,
+	}
+}
+
+func (p *Pprof) Run(ctx context.Context) error {
+	if !p.enabled {
+		return nil
+	}
+
+	server := &http.Server{
+		Addr:    "localhost" + p.port,
+		Handler: http.DefaultServeMux,
+	}
+
+	started := make(chan struct{})
+	go func() {
+		started <- struct{}{}
+		err := server.ListenAndServe()
+		if err != nil && err != http.ErrServerClosed {
+			p.log.Errorw("Pprof server error", "err", err)
+		}
+	}()
+
+	go func() {
+		<-ctx.Done()
+		server.Shutdown(context.Background())
```

Comment:
See how `http` handles this, we need pprof to be similar to that.

Reply:
Not sure what you exactly meant but I made it a little bit simpler and added error handling for shut downing.

Reply:
http propagates the shutdown error to the caller, maybe we should do the same here as well.

Reply:
changed - now error on shutdown is returned